x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/FieldAttribute.java

```diff
@@ -29,6 +29,7 @@
     private final FieldAttribute nestedParent;
     private final String path;
     private final EsField field;
+    private final String indexName;
```

Comment:
This doesn't work since a field can be mapped across multiple indices. Since the attributes (which in this case are mapped as fields) are likely the result of a virtual mapping, the notion of an index will not hold and end up being as an 'exceptional' code path that is sometimes used.

Reply:
True, `indexName` is a poor naming choice since it contains the index pattern from the SQL query, not the actual source index/indices. Its original name has been derived from `EsIndex`, where it's extracted from.
Since we can't know the actual index/indices origin, I guess more appropriate might be something in the lines of `indexPattern`.